src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/controllers/OrderController.java

```diff
@@ -65,6 +63,36 @@ List<OrderResponseDTO> getOrders(@RequestParam(value = "page", defaultValue = "0
         return orderServiceFacade.getAllOrdersBySearchingData(build);
     }
 
+
+    @PatchMapping("/{id}/confirm")
+    OrderResponseDTO confirmOrder(@PathVariable Long id) {
+        return orderServiceFacade.proceedOrderAfterConfirm(id);
+    }
+
+    @PatchMapping("/{id}/reject")
+    ResponseEntity<String> rejectOrder(@PathVariable Long id) {
+        orderServiceFacade.rejectOrder(id);
+        return ResponseEntity.ok().body("Order was rejected");
+    }
+
+    @PatchMapping("/{codeId}/cancel")
+    ResponseEntity<String> cancelOrder(@PathVariable String codeId) {
+        orderServiceFacade.cancelOrder(codeId);
+        return ResponseEntity.ok().body("Your order was cancel");
+    }
+
+    @PatchMapping("/{codeId}/payment")
+    ResponseEntity<String> payForOrder(@PathVariable String codeId) { //tmp controller to simulation situation when client pay for Order
+        orderServiceFacade.proceedOrderAfterPayment(codeId);
+        return ResponseEntity.ok().body("Your payment was success");
+    }
+
+    @PostMapping("/shipment")
+    ResponseEntity<Void> updateStatusFromWarehouse(@RequestBody @Valid OrderDataFromWarehouseDTO orderDataFromWarehouseDTO) {
```

Comment:
OrderDataFromWarehouseDTO -> to powinna być inna klasa, niż to, co wpada do domeny
Generalnie nasza domena powinna być niezależna od tego, z jakim system magazynowym się dogadaliśmy i co oni wysyłają. Potrzebujemy więc jakiejś warstwy mapującej